src/components/reports/chartSpecs.js

```diff
@@ -0,0 +1,212 @@
+const graphColorType = {
+  mediumPink: '#cc79a7',
+  mediumYellow: '#f0e442',
+  mediumBlue: '#0072b2',
+  lightBlueGreen: '#44aa99',
+  darkPurple: '#342288',
+  black: '#000000',
+};
+
+function returnGradeCount(scores) {
+  // gradecount should be an obj of {{grade:{} count}}
+  let gradeCount = [
+    { grade: 'Pre-K', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: 'T-K', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: 'Kindergarten', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '1', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '2', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '3', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '4', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '5', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '6', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '7', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '8', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '9', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '10', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '11', support_levels: [0, 0, 0], totalStudents: 0 },
+    { grade: '12', support_levels: [0, 0, 0], totalStudents: 0 },
+  ];
+  for (let score of scores.value) {
+    let gradeCounter = gradeCount.find((grade) => grade.grade === score?.user?.grade?.toString());
+    if (gradeCounter) {
+      if (score?.scores?.support_level === 'Needs Extra Support' && gradeCounter) {
+        gradeCounter.support_levels[0]++;
+        gradeCounter.totalStudents++;
+      } else if (score?.scores?.support_level === 'Needs Some Support' && gradeCounter) {
+        gradeCounter.support_levels[1]++;
+        gradeCounter.totalStudents++;
+      } else if (score?.scores?.support_level === 'At or Above Average' && gradeCounter) {
+        gradeCounter.support_levels[2]++;
+        gradeCounter.totalStudents++;
+      } else {
+        // score not counted (support level null)
+      }
+    }
+  }
+
+  return gradeCount;
+}
+
+function returnValueByIndex(index, grade, mode) {
+  if (index >= 0 && index <= 2) {
+    // 0 => needs extra support
+    // 1 => needs some support
+    // 2 => at or above average
+    let valsByIndex = [
+      { group: 'Needs Extra Support' },
+      { group: 'Needs Some Support' },
+      { group: 'At or Above Average' },
+    ];
+    let value;
+    if (mode === 'percentage') {
+      value = {
+        category: grade.grade,
+        group: valsByIndex[index].group,
+        color: valsByIndex[index].color,
+        value: (grade?.support_levels[index] / grade.totalStudents) * 100,
+      };
+    }
+    if (mode === 'count') {
+      value = {
+        category: grade.grade,
+        group: valsByIndex[index].group,
+        color: valsByIndex[index].color,
+        value: grade?.support_levels[index],
+      };
+    }
+    return value;
```

Comment:
Suggestion: return the value in each `if` block rather than saving to the `value` variable. Also, it doesn't appear that `valsByIndex[index].color` is defined so I removed those lines in this suggestion.

```suggestion
    if (mode === 'percentage') {
      return {
        category: grade.grade,
        group: valsByIndex[index].group,
        value: (grade?.support_levels[index] / grade.totalStudents) * 100,
      };
    }
    if (mode === 'count') {
      return {
        category: grade.grade,
        group: valsByIndex[index].group,
        value: grade?.support_levels[index],
      };
    }
    return value;
```